excel_import_export/wizard/import_xlsx_wizard.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+
+from odoo import _, api, fields, models
+from odoo.exceptions import RedirectWarning, ValidationError
+
+
+class ImportXLSXWizard(models.TransientModel):
+    """This wizard is used with the template (xlsx.template) to import
+    xlsx template back to active record"""
+
+    _name = "import.xlsx.wizard"
+    _description = "Wizard for importing excel"
+
+    import_file = fields.Binary(string="Import File (*.xlsx)")
+    filename = fields.Char("Import File Name")
+    template_id = fields.Many2one(
+        "xlsx.template",
+        string="Template",
+        required=True,
+        ondelete="cascade",
+        domain=lambda self: self._context.get("template_domain", []),
```

Comment:
This template filtering doesn't seem to work.
When you have `sale.order` and `account.move` (invoice) templates, this should filter out specific templates according to the model, but instead shows all of them on both Sales and Invoices (see images below)

While testing, sometimes I would also get `No template found` Validation Error that comes from `default_get` method. 

There must be something wrong with context generation and passing.

![image](https://github.com/user-attachments/assets/d6a774b9-e42e-4eea-ab10-c6061dcaa095)

In debug mode, you can see that the domain isn't even set.
![image](https://github.com/user-attachments/assets/63332d5e-c73e-4a11-9688-2b812b7d47d1)

It would be very appreciated if you would add a test case for this. Mainly, checking if `template_domain` is generated and passed correctly.